Only serialize installed fonts attributes for WebCore::AttributedString

gavin-apple opened the pull request:
#### 4fc463d4e6296b020cf1243b712463b7a4b8b0da
<pre>
Only serialize installed fonts attributes for WebCore::AttributedString
<a href="https://bugs.webkit.org/show_bug.cgi?id=299174">https://bugs.webkit.org/show_bug.cgi?id=299174</a>
&lt;<a href="https://rdar.apple.com/problem/161175544">rdar://problem/161175544</a>&gt;

Reviewed by Alex Christensen.

This change modifies our WebCore::AttributedString serialzation to only support
installed fonts.

Test: Tools/TestWebKitAPI/Tests/WebKitCocoa/NSAttributedStringWebKitAdditions.mm
* Source/WebCore/Configurations/AllowedSPI.toml:
* Source/WebCore/PAL/pal/spi/cf/CoreTextSPI.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::toNSObject):
(WebCore::extractValue):
(WebCore::AttributedString::FontWrapper::createFromIPCData):
(WebCore::AttributedString::FontWrapper::postScriptName const):
(WebCore::AttributedString::FontWrapper::pointSize const):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NSAttributedStringWebKitAdditions.mm:
(TEST(NSAttributedStringWebKitAdditions, InstalledFont)):

Canonical link: <a href="https://commits.webkit.org/300681@main">https://commits.webkit.org/300681@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95b9760b5fabcfd402cc906cca98c795f2e0b0a0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122901 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42613 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33303 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129539 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75007 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e7a911b1-48bc-4499-9de4-6a6a983f40be) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124777 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43332 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51206 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93449 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62012 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ceb63aed-b1bc-4f55-96f1-9fcb14739534) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125853 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34545 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110009 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74042 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/72aaf893-2c43-4178-b26a-813cc8b04636) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33524 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28166 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73030 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104247 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28378 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132269 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49846 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37946 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101937 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50223 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106222 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101801 "Found 4 new API test failures: TestWTF:CompletionHandlerDeathTest.MainThreadHandlerOnThreadAssertsDeathTest, WebKitGTK/TestInspector:/webkit/WebKitWebInspector/manual-attach-detach, TestWTF:CompletionHandlerDeathTest.ConstructionThreadHandlerOnThreadAssertsDeathTest, WebKitGTK/TestWebKitNetworkSession:/webkit/WebKitNetworkSession/proxy (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25982 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47160 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25349 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/46618 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49704 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55456 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49174 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52522 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50856 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4a434fc20c62e52d42c40a949a311d725dc78806. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51004

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7093ed61cd2fbcb18c9aed949f19d17ab4eb53db. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51004

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/40d5e21795c10b88ba67af59365e98b95f29ff5d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51004

gavin-apple committed:
Only serialize installed fonts attributes for WebCore::AttributedString
https://bugs.webkit.org/show_bug.cgi?id=299174
<rdar://problem/161175544>

Reviewed by Alex Christensen.

This change modifies our WebCore::AttributedString serialzation to only support
installed fonts.

Test: Tools/TestWebKitAPI/Tests/WebKitCocoa/NSAttributedStringWebKitAdditions.mm
* Source/WebCore/Configurations/AllowedSPI.toml:
* Source/WebCore/PAL/pal/spi/cf/CoreTextSPI.h:
* Source/WebCore/editing/cocoa/AttributedString.h:
* Source/WebCore/editing/cocoa/AttributedString.mm:
(WebCore::toNSObject):
(WebCore::extractValue):
(WebCore::AttributedString::FontWrapper::createFromIPCData):
(WebCore::AttributedString::FontWrapper::postScriptName const):
(WebCore::AttributedString::FontWrapper::pointSize const):
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/NSAttributedStringWebKitAdditions.mm:
(TEST(NSAttributedStringWebKitAdditions, InstalledFont)):

Canonical link: https://commits.webkit.org/300681@main

webkit-commit-queue commented:
Committed 300681@main (4fc463d4e629): <https://commits.webkit.org/300681@main>

Reviewed commits have been landed. Closing PR #51004 and removing active labels.
